### PR TITLE
mtie should output the correct rank and size for the output operator.

### LINKED ISSUE #725: [BUG] Transform Ops have incorrect rank and size
When a user creates a transform op then queries the rank and size of that op it returns -1 for rank and 0 for size.   

The cause of this is that we return an mtie for transform ops here:

https://github.com/NVIDIA/MatX/blob/main/include/matx/core/tensor.h#L285-L296

Mtie currently returns matxNoRank for rank and 0 for size here:

https://github.com/NVIDIA/MatX/blob/main/include/matx/core/tie.h#L84-L92

This causes issues if you try to then pass that transform into other ops.

### PR DESCRIPTION
**Pull request by luitjens**
Fixes #725

### CODE REVIEW

File: test/00_transform/Norm.cu [hunk 1]
```diff
@@ -85,8 +85,16 @@ TYPED_TEST(NormTestFloatTypes, VectorL1)
...
+  auto redOp = vector_norm(this->in_v, NormOrder::L1);
+
+  EXPECT_TRUE(redOp.Rank() == this->out_v.Rank());
+  for(int i = 0; i < redOp.Rank(); i++) {
+    EXPECT_TRUE(redOp.Size(i) == this->out_v.Size(i));
+  }
+
   // example-begin vector-norm-test-1
-  (this->out_v = vector_norm(this->in_v, NormOrder::L1)).run(this->exec);
+  (this->out_v = redOp).run(this->exec);
```
Comment (cliffburdick) [inline test/00_transform/Norm.cu:97]: This changes the example in the docs so it no longer shows the norm. This should probably be a separate test without a docs example

File: test/00_transform/Norm.cu [hunk 2]
```diff
@@ -102,8 +110,15 @@ TYPED_TEST(NormTestFloatTypes, VectorL2)
...
+  
+  auto redOp = vector_norm(this->in_v, NormOrder::L2);
+  
+  EXPECT_TRUE(redOp.Rank() == this->out_v.Rank());
+  for(int i = 0; i < redOp.Rank(); i++) {
+    EXPECT_TRUE(redOp.Size(i) == this->out_v.Size(i));
+  }
   // example-begin vector-norm-test-2
-  (this->out_v = vector_norm(this->in_v, NormOrder::L2)).run(this->exec);
+  (this->out_v = redOp).run(this->exec);
```
Comment (cliffburdick) [inline test/00_transform/Norm.cu:121]: Same

File: test/00_transform/Norm.cu (hunk 1 above)
Comment (luitjens) [reply test/00_transform/Norm.cu]: fixed

File: test/00_transform/Norm.cu (hunk 2 above)
Comment (luitjens) [reply test/00_transform/Norm.cu:124]: fixed

**Review by cliffburdick — APPROVED** [review]

**Pull request merged**